clusterDeployer.py

```diff
@@ -285,7 +285,17 @@ def create_masters(self):
         procs = setup_vms(self._cc["masters"], os.path.join(os.getcwd(), f"{infra_env}.iso"), self.args.images_path)
 
         print("Waiting for all hosts to be in \'known\' state")
-        self._wait_known_state(e["name"] for e in self._cc["masters"])
+
+        names = { e["name"] for e in self._cc["masters"] }
+        self._init_hosts_state(names)
+        while True:
+            if self._check_known_state(names):
+              break;
+            time.sleep(1)
+
+            if any(not p.is_alive() for p in procs):
```

Comment:
I'm thinking about this one. I don't know about how it behaves on your system, but the thread that wraps the running VM exits when the VM reboots once. two VMs will reboot first, then 1 more.

What's the behavior on your side?

Reply:
I'm not sure I did not track exactly the moment the thread completes on successful execution. What I'm trying to do here is catching the situation where virt_install fails (exits with error) before being actually able to do anything relevant - e.g. because the partition containing the images is too small.
That happens well before reaching the known status. AFAICS the reboot happens quite later.